openmdao/components/eq_constraints_comp.py

```diff
@@ -0,0 +1,249 @@
+"""Define the EqualityConstraintsComp class."""
+
+from __future__ import print_function, division, absolute_import
+
+from numbers import Number
+from six import iteritems
+
+import numpy as np
+
+from openmdao.core.explicitcomponent import ExplicitComponent
+
+
+class EqualityConstraintsComp(ExplicitComponent):
+    """
+    A component that computes the difference between two inputs to test for equality.
+
+    Attributes
+    ----------
+    _output_vars : dict
+        Cache the data provided during `add_eq_output`
+        so everything can be saved until setup is called.
+    """
+
+    def __init__(self, name=None, eq_units=None, lhs_name=None, rhs_name=None, rhs_val=0.0,
+                 use_mult=False, mult_name=None, mult_val=1.0, add_constraint=False,
+                 **kwargs):
+        r"""
+        Initialize an EqualityConstraintsComp, optionally add an output constraint to the model.
+
+        The EqualityConstraintsComp is a bit like IndepVarComp in that it allows for the
```

Comment:
I don't like the comparison to IndepVarComp. I know the API is similar, but the usage is not. Plus the wording is weird, because it implies that IndepVarComp uses a similar equation. 

```Initialize an EqualityConstraintsComp with one ore more outputs, that can optionally also be added as constraint to the driver.

The EQConstraintComp is similar to the BalanceComp, in that works with the following equations: ...```

Reply:
changed in EQComp and in original BalanceComp